.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,20 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - '*'
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+      - uses: actions/checkout@v3
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.50.1
+          args: "--out-${NO_FUTURE}format colored-line-number"
```

Comment:
Kindly add the following args as well. This is what we do for the control plane project
--timeout 5m --enable gosec
It will make sure the linter doesn't stuck forever and the second argument is to perform code security checks. Be default gosec is not included.

Reply:
thanks @miazamrai updated the PR accordingly